### PR TITLE
program: support attach to BPF iterators.

### PR DESCRIPTION
**Pull request by WeiZhang555**
Add new function `attach_iter` to support BPF iter program, since iterator requires a map fd as attach options, general `attach` function can't work, we must add standalone `attach_iter` function.

### CODE REVIEW

File: libbpf-rs/src/program.rs [hunk 1]
```diff
@@ -772,6 +772,28 @@ impl Program {
         )
     }
 
+    /// Attach this program to a
+    /// [BPF iter](https://www.kernel.org/doc/html/latest/trace/tracepoints.html).
+    /// The entry point of the program must be defined with `SEC("iter")` or `SEC("iter.s")`.
+    pub fn attach_iter(&mut self, map_fd: u32) -> Result<Link> {
```
Comment (danielocfb) [inline libbpf-rs/src/program.rs:778]: We recently started using Rust's [`fd` abstractions](https://doc.rust-lang.org/std/os/fd/index.html), `BorrwedFd` and `OwnedFd`. Can we use them here please?

File: libbpf-rs/src/program.rs [hunk 2]
```diff
@@ -772,6 +772,28 @@ impl Program {
         )
     }
 
+    /// Attach this program to a
+    /// [BPF iter](https://www.kernel.org/doc/html/latest/trace/tracepoints.html).
```
Comment (danielocfb) [inline libbpf-rs/src/program.rs:776]: This link destination does not seem on topic. Is https://www.kernel.org/doc/html/latest/bpf/bpf_iterators.html more fitting?

**Review by danielocfb — COMMENTED** [review]
Overall this seems like a reasonable addition, thanks! But can you please add a unit test for this functionality? It helps verify that everything is wired up correctly and illustrates usage.

File: libbpf-rs/src/program.rs (hunk 2 above)
Comment (WeiZhang555) [reply libbpf-rs/src/program.rs:776]: Yes, you're right. I copied/pasted a wrong link.

I will fix it in a few days once after I come back from  my vacation.

File: libbpf-rs/src/program.rs (hunk 1 above)
Comment (WeiZhang555) [reply libbpf-rs/src/program.rs]: Fixed. @danielocfb

**Comment by danielocfb** [comment]
Thanks for the update! I merged the change with some tiny modifications.

**Comment by WeiZhang555** [comment]
> Thanks for the update! I merged the change with some tiny modifications.

@danielocfb Thanks!

**Pull request closed without merging**